src/pyqasm/visitor.py

```diff
@@ -1939,8 +1949,34 @@ def _visit_function_call(
 
         return return_value, result
 
-    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> None:
-        pass
+    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> list[qasm3_ast.Statement]:
+        """Visit a while loop statement element."""
+        result: list[qasm3_ast.Statement] = []
+        # Check for classical register in while condition
+        if self.classical_register_in_expr(statement.while_condition):
+            raise_qasm3_error(
+                "Cannot unroll while loops with quantum measurement in the condition."
+            )
+        iterations = 0
+        loop_limit = self._loop_limit
+        self._curr_scope += 1
```

Comment:
Hi @luffy-orf , any reason why you removed [the addition of scopes](https://github.com/qBraid/pyqasm/pull/199/commits/a9b650a958de124bbe12b356d645286936198f38#diff-802ca57af29fc92ca8eb2f536fb5217c61c874c4c35bd39b7fd82512ee0e9a84L1969) from here and the break / continue statements?